docs/source/api.rst

```diff
@@ -115,6 +115,16 @@ Connectivity and Cycles
    retworkx.cycle_basis
    retworkx.digraph_find_cycle
 
+.. _approximations:
+
+Approximations and Heuristics
+-----------------------------
+
+.. autosummary::
+   :toctree: stubs
+
+   retworkx.metric_closure
+
```

Comment:
Although i like this new doc section, i'm not sure if this is the best place for `metric_closure`. i would suggest the other-algorithms section.

Reply:
I just copied this from where networkx puts it, but I think that's only because the steiner tree function is there with it, I'll move it to the other algorithms section

Reply:
Done in: https://github.com/Qiskit/retworkx/pull/390/commits/1641c8e4c3fc4207c308e9126a0c501e0f37eef0